src/node_buffer.cc

```diff
@@ -1193,13 +1221,29 @@ void Swap32(const FunctionCallbackInfo<Value>& args) {
   THROW_AND_RETURN_UNLESS_BUFFER(env, args.This());
   SPREAD_ARG(args.This(), ts_obj);
 
-  for (size_t i = 0; i < ts_obj_length; i += 4) {
-    SWAP_BYTES(ts_obj_data, i, i + 3);
-    SWAP_BYTES(ts_obj_data, i + 1, i + 2);
+  uint32_t* data32 = reinterpret_cast<uint32_t*>(ts_obj_data);
+  size_t len32 = ts_obj_length / 4;
+
+  for (size_t i = 0; i < len32; i++) {
+	  data32[i] = BSWAP_INTRINSIC_4(data32[i]);
   }
   args.GetReturnValue().Set(args.This());
 }
 
+void Swap64(const FunctionCallbackInfo<Value>& args) {
+	Environment* env = Environment::GetCurrent(args);
+	THROW_AND_RETURN_UNLESS_BUFFER(env, args.This());
+	SPREAD_ARG(args.This(), ts_obj);
+
+	uint64_t* data64 = reinterpret_cast<uint64_t*>(ts_obj_data);
+	size_t len32 = ts_obj_length / 8;
```

Comment:
Mind adding a `CHECK_EQ(ts_obj_length % 8, 0)` just above this? Same (but with 4) for Swap32.